### ISSUE TITLE
Dangling commits

### ISSUE THREAD

**Issue by janezicmatej** [zegl/extremely-linear#6]
While playing around with this great tool I've found another thing.

Amending commits and rerunning `git linearize --format %07d0` (default post-commit hook) will produce dangling commits, which will render `shit` commands less useful as e.g. `shit checkout 22` will not match a unique short ref anymore.

Since this is expected git behaviour, this isn't exactly an issue and can be solved quite easily with something like

```sh
git reflog expire --expire-unreachable=now --all
git gc --prune=now
```

but maybe a helper function like `shit plunge` could be implemented that does this (or something similar)

**Comment by zegl** [comment]
> `shit plunge`

LOL! Hahaha this is honestly a good idea. Alternatively, `shit` should maybe only consider commits that are reachable from HEAD? Or some other active branch.

I'll do some experimentation and see which idea that is the best one. I'd like to avoid deleting git objects if we can... it seems risky.

**Comment by janezicmatej** [comment]
Commits only reachable from HEAD sounds like a great idea!

**Comment by zegl** [comment]
I think it works! I've pushed a fix in 000005509899d25b780db196379965c69e2c0685 :-)

Let me know if it works for you. :-)

**Comment by janezicmatej** [comment]
Yup works. Closing this issue now.
